src/leiningen/jupyter.clj

```diff
@@ -1,26 +1,33 @@
 (ns leiningen.jupyter
-  (:require [leiningen.core.main]
-            [leiningen.jupyter.extension :refer [install-and-enable-extension]]
-            [leiningen.jupyter.kernel :refer [install-kernel
-                                              run-kernel
-                                              kernel-installed?]]
-            [leiningen.jupyter.params :as params])
+  (:require
+   [leiningen.core.main :as leiningen.main]
+   [leiningen.jupyter.extension :refer [install-and-enable-extension]]
+   [leiningen.jupyter.kernel :refer [install-kernel kernel-installed?
+                                     run-kernel]]
+   [leiningen.jupyter.params :as params]
+   [leiningen.jupyter.utils :as utils])
 
-  (:import [org.apache.commons.exec CommandLine
-                                    DefaultExecutor
-                                    PumpStreamHandler]))
+  (:import
+   [org.apache.commons.exec CommandLine DefaultExecutor PumpStreamHandler]))
 
 (defn get-jupyter-command [jupyter sub-command jupyter-arguments]
   (let [all-arguments (into [sub-command] jupyter-arguments)
         add-args #(.addArgument %1 %2)]
       (reduce add-args (new CommandLine jupyter) all-arguments)))
 
+(defn ^:private jupyter-start-sub-command
+  [jupyter-exe]
+  (let [jupyter-version (utils/jupyterlab-version jupyter-exe)]
+    (if (= 4 jupyter-version)
```

Comment:
Would it make sense to use >=?
```suggestion
    (if (>= 4 jupyter-version)
```
If so, maybe we'll need to modify those extension `defmulti`s to have only `4` and `else` as _options_
WDYT?

Reply:
![image](https://github.com/nubank/lein-jupyter/assets/609780/a5969c30-9af2-4827-8de6-98e249c6e743)

Reply:
Yes, I think so, but for the `defmult`, we probably would need logic to handle versions `<=3` and `>=4` 🤔

Reply:
@jsanchezdev  I have made some changes, WDYT?

Reply:
LGTM, thanks!

Reply:
Good catch!